src/cli/Cli.ts

```diff
@@ -404,17 +404,68 @@ export class Cli {
       const commandTypes = commandInfo.command.types;
       if (commandTypes) {
         minimistOptions.string = commandTypes.string;
-        minimistOptions.boolean = commandTypes.boolean;
+
+        // Only treat those options as booleans that are included as arg
+        // Otherwise unincluded options will default to false.
+        minimistOptions.boolean = commandTypes.boolean?.filter(optionName => args.some(arg => `--${optionName}` === arg || `-${optionName}` === arg));
       }
+
       minimistOptions.alias = {};
       commandInfo.options.forEach(option => {
+        Cli.log(JSON.stringify(option));
         if (option.short && option.long) {
           (minimistOptions.alias as any)[option.short] = option.long;
         }
       });
     }
 
-    return minimist(args, minimistOptions);
+    const curatedArgs = this.getCuratedArgsList(args, commandInfo);
+
+    return minimist(curatedArgs, minimistOptions);
+  }
+
+  /**
+   * Curates arguments before passing them into minimist.
+   * Currently only boolean values are checked and fixed.
+   */
+  private getCuratedArgsList(args: string[], commandInfo: CommandInfo | undefined): string[] {
```

Comment:
Let's change the name of this method to be clearer. "curating" seems vague. Since it's private, we can always rename it without any impact should its scope change.